### PR TITLE
Add postgres support for JDBIHistory

### PR DESCRIPTION
**Pull request by mikebell90**
* adds migrations (single changeset) for PG  …
* refactors HistoryJDBI into an interface, with an abstract
base class for shared code (limit offset using standard SQL idiom)
and concrete subclasses for Postgres and MySQL.
* SingularityMappers can choose to use either f_user or user column - this is needed because Postgres considers user a reserved word.
* Modify SingularityHistoryModule to handle the new bindings
* JDBIHistoryManager - add debug logging at TRACE level.

### CODE REVIEW

**Comment by mikebell90** [comment]
@ssalinas I deliberately split everything so.no migration or compatibility concerns would arise with mysql. They were 99.9999% compatible other than the user column and the offset limit syntax. But I thought this was cleaner in the end

File: SingularityService/src/main/java/com/hubspot/singularity/data/history/HistoryJDBI.java [hunk 1]
```diff
@@ -1,19 +1,7 @@
 package com.hubspot.singularity.data.history;
```
Comment (mikebell90) [inline SingularityService/src/main/java/com/hubspot/singularity/data/history/HistoryJDBI.java]: This class now just an interface

File: SingularityService/src/main/java/com/hubspot/singularity/data/history/JDBIHistoryManager.java [hunk 2]
```diff
@@ -42,14 +42,26 @@ public JDBIHistoryManager(HistoryJDBI history, Transcoder<SingularityTaskHistory
   @Timed
   public List<SingularityTaskIdHistory> getTaskIdHistory(Optional<String> requestId, Optional<String> deployId, Optional<String> runId, Optional<String> host, Optional<ExtendedTaskState> lastTaskStatus, Optional<Long> startedBefore,
       Optional<Long> startedAfter, Optional<Long> updatedBefore, Optional<Long> updatedAfter, Optional<OrderDirection> orderDirection, Optional<Integer> limitStart, Integer limitCount) {
-    return history.getTaskIdHistory(requestId, deployId, runId, host, lastTaskStatus, startedBefore, startedAfter, updatedBefore, updatedAfter, orderDirection, limitStart, limitCount);
+
+    List<SingularityTaskIdHistory> taskIdHistoryList =  history.getTaskIdHistory(requestId, deployId, runId, host, lastTaskStatus, startedBefore, startedAfter, updatedBefore, updatedAfter, orderDirection, limitStart, limitCount);
+    if (LOG.isTraceEnabled()) {
```
Comment (mikebell90) [inline SingularityService/src/main/java/com/hubspot/singularity/data/history/JDBIHistoryManager.java]: I added logging at trace level for the sql stuff. Seems a useful thing

File: SingularityService/src/main/java/com/hubspot/singularity/data/history/MySQLHistoryJDBI.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+import org.skife.jdbi.v2.sqlobject.Bind;
+import org.skife.jdbi.v2.sqlobject.SqlQuery;
+import org.skife.jdbi.v2.sqlobject.SqlUpdate;
+import org.skife.jdbi.v2.sqlobject.customizers.Define;
+import org.skife.jdbi.v2.sqlobject.stringtemplate.UseStringTemplate3StatementLocator;
+
+@UseStringTemplate3StatementLocator
+public abstract class MySQLHistoryJDBI extends AbstractHistoryJDBI {
+
+  @SqlUpdate("INSERT INTO requestHistory (requestId, request, createdAt, requestState, user, message) VALUES (:requestId, :request, :createdAt, :requestState, :user, :message)")
```
Comment (mikebell90) [inline SingularityService/src/main/java/com/hubspot/singularity/data/history/MySQLHistoryJDBI.java]: Pretty much the old jdbi history class minus the common code

**Comment by ssalinas** [comment]
Hey @mikebell90 thanks for the PR! Was a busy week here at HubSpot, I will try and take a closer look at this early next week 👍

**Comment by mikebell90** [comment]
Cool! Thanks

**Comment by mikebell90** [comment]
@ssalinas Yeah that was dumb of me. I went on memory of MySQL that it accepted both in that order ;)

**Comment by ssalinas** [comment]
This has been pretty stable in our setup so far. I'd say let's 🚢 . We'll get this in our production branch and merged this week. Can try to cut a release hopefully this week as well

**Comment by mikebell90** [comment]
Very cool!

**Pull request merged**